### DISCUSSION TITLE
Want a built-in plugin for unocss

### DISCUSSION THREAD

**Discussion by xhebox** [remix-run/remix#4336] [Proposals]
Some of tailwindcss/unocss/windicss features requires source file transformation, e.g. https://uno.antfu.me/?s=guide:transformer-variant-group.

```
<div class="hover:(bg-gray-400 font-medium) font-(light mono)"/>

<div class="hover:bg-gray-400 hover:font-medium font-light font-mono"/>
```

This is an extremely convenient syntax sugar. But it is very buggy/hard to use their CLI to scan all files and get the pre-generated sources. This should be integrated into esbuild instead.

**Comment by sergiodxa** [comment]
The problem of this is that you want this plugin, and then another person wants another plugin, and if Remix keeps adding every plugin for every tool the code will become too complex and the build times too slow.

Ideally this things runs outside Remix so you're responsible, you could for example keep an `src` folder with your code using unocss, run a CLI to process that and put the output on `app` where then Remix can pick the code and do its build.

Another option is that Remix enables some plugins system but the team has mentioned before that it's not something they want to do yet because it'll make changing the compiler a breaking change and they're not ready for marrying with the current compiler.

**Reply by xhebox** [reply]
I understand. So plugins are hardcoded somehow? That's a sad story..

I've tried a little bit with esbuild plugin systems. IIRC, it is indeed hard to construct a transformer plugin system as mdast.

I guess that it is more correct to let remix compiler accepts something like `[]Transformer`, not utilizing esbuild plugin system. It is flexible enough to enable transformations on various file types, especially `jsx/tsx`.
